cmd/kubeadm/app/preflight/checks.go

```diff
@@ -986,6 +1019,9 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.JoinConfigura
 				HTTPProxyCheck{Proto: "https", Host: ipstr},
 			)
 			if ip := netutils.ParseIPSloppy(ipstr); ip != nil {
+				if netutils.IsIPv4(ip) {
+					checks = addIPv4Checks(checks)
+				}
```

Comment:
It seems that we can only do IPv4 check or IPv6 check for `RunJoinNodeChecks`, since the `APIServerEndpoint` can only be an IPv4 or IPv6 IP. How about dual stack check for `RunJoinNodeChecks`? Can we do it just like `RunInitNodeChecks`?

I +1 for this feature, but it seems we need to do more :)
And we need some e2e tests for this feature.

/approve
/hold  
free to cancel hold if needed